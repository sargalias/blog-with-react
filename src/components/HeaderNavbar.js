
import React from 'react';
import NavLink from './NavLink';


const HeaderNavbar = () => (
  <nav className="navbar navbar-expand-lg navbar-light bg-light" id="navbar">
    <div className="container">
      <a className="navbar-brand text-primary" href="/">
        <div className="large">Blog</div>
        <div className="small">with React</div>
      </a>
      <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
        <span className="navbar-toggler-icon"></span>
      </button>

      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav mr-auto">
          <NavLink link='/' label='Home' isActive={true} />
          <NavLink link='/new' label='Add New' />
          <NavLink link='/404' label='404' />
          <NavLink link='/single' label='Single' />
        </ul>
        <ul className="navbar-nav ml-auto">
          <NavLink link='/dashboard' label='Dashboard' />
          <NavLink link='/login' label='Log In' />
          <NavLink link='/logout' label='Log Out' />
        </ul>
      </div>
    </div>
  </nav>
);

export default HeaderNavbar;
