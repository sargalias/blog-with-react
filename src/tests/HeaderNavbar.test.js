import React from 'react';
import { shallow } from 'enzyme';
import HeaderNavbar from '../components/HeaderNavbar';

test('HeaderNavbar should render correctly', () => {
  const wrapper = shallow(<HeaderNavbar />);
  expect(wrapper).toMatchSnapshot();
});
