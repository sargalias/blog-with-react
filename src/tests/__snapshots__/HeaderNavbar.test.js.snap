// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavbar should render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
  id="navbar"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand text-primary"
      href="/"
    >
      <div
        className="large"
      >
        Blog
      </div>
      <div
        className="small"
      >
        with React
      </div>
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <NavLink
          isActive={true}
          label="Home"
          link="/"
        />
        <NavLink
          label="Add New"
          link="/new"
        />
        <NavLink
          label="404"
          link="/404"
        />
        <NavLink
          label="Single"
          link="/single"
        />
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <NavLink
          label="Dashboard"
          link="/dashboard"
        />
        <NavLink
          label="Log In"
          link="/login"
        />
        <NavLink
          label="Log Out"
          link="/logout"
        />
      </ul>
    </div>
  </div>
</nav>
`;
